esmvaltool/preprocessor/_mask.py

```diff
@@ -129,6 +136,29 @@ def mask_landsea(cube, fx_files, mask_out):
     return cube
 
 
+def mask_landseaice(cube, fx_files, mask_out):
+    """Apply a land/ice mask"""
+    # mask_out: is either 'land' or 'ice'
+
+    # sftgif is the only one so far
+    if fx_files:
+        fx_cubes = {}
+        for fx_file in fx_files:
+            fx_root = os.path.basename(fx_file).split('_')[0]
+            fx_cubes[fx_root] = iris.load_cube(fx_file)
+
+        if ('sftgif' in fx_cubes.keys() and
+                _check_dims(cube, fx_cubes['sftgif'])):
+            landice_mask = _get_fx_mask(fx_cubes['sftgif'].data, mask_out,
+                                        'sftgif')
+            cube.data = _apply_fx_mask(landice_mask, cube.data)
+            logger.debug("Applying landsea-ice mask: sftgif")
+    else:
+        logger.error("Landsea-ice mask could not be found ")
```

Comment:
Change this to a warning. If it is really an error if no fx file is available, this should raise an exception already when parsing the recipe.